src/application/communication/Broker.java

```diff
@@ -279,4 +317,94 @@ private void startListeningTCP()
         }).start();
 
     }
+
+    private void receiveTCP(byte[] data)
+    {
+        ByteBuffer buffer = ByteBuffer.allocate(data.length);
+        buffer.order(ByteOrder.LITTLE_ENDIAN);
+        buffer.put(data);
+        buffer.rewind();
+
+        if (buffer.get() != 1) // start of heading
+        {
+            return;
+        }
+        byte value = buffer.get();
+        Opcodes opcode = Opcodes.fromInteger(value);
+
+        buffer.get(); // start of text
+
+        switch (opcode)
+        {
+            case REQUEST_BULLET:
+            {
+                handleBulletCreation(buffer);
+            }
+            case NEW_PLAYER:
+            {
+                handleNewPlayer(buffer);
+            }
+            case DISCONNECTED_PLAYER:
+            {
+                handleDisconnectedPlayer(buffer);
+            }
+            default:
+                break;
+
+        }
+
+    }
+
+    /**
+     * Handles new players created by the server.
+     * 
+     * @param input
+     *            The ByteBuffer that handles reading of data send from the server.
+     */
+    private void handleDisconnectedPlayer(ByteBuffer input) // Should probably tell GameClient about the new player instead
+    {
+        int playerId = input.getInt();
+        int objectId = input.getInt();
+        map.removeGameObject(objectId);
+
+    }
+
+    /**
+     * Handles new players created by the server.
+     * 
+     * @param input
+     *            The ByteBuffer that handles reading of data send from the server.
+     */
+    private void handleNewPlayer(ByteBuffer input) // Should probably tell GameClient about the new player instead
+    {
+        GameObjectDAO data = new GameObjectDAO();
+        int playerId = input.getInt();
+        data.objectId = input.getInt();
+        data.x = input.getDouble();
+        data.y = input.getDouble();
+        data.width = input.getDouble();
+        data.height = input.getDouble();
+        map.addGameObject(data);
+    }
+
+    /**
+     * Handles new bullets created by the server.
+     * 
+     * @param input
+     *            The ByteBuffer that handles reading of data send from the server.
+     */
+    private void handleBulletCreation(ByteBuffer input)
+    {
+        GameObjectDAO data = new GameObjectDAO();
+        data.x = input.getDouble();
+        data.y = input.getDouble();
+        data.height = input.getDouble();
+        data.velocityX = input.getDouble();
+        data.velocityY = input.getDouble();
+        data.bulletType = Bullet.Type.values()[input.getInt()];
```

Comment:
this is very very expensive. instead make a static array in the enum class from which you can get opcode by index